libs/core/src/main/java/org/opensearch/core/common/io/StreamsUtil.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.core.common.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Util library for reading streams
+ *
+ * @opensearch.internal
+ */
+public final class StreamsUtil {
+
+    private StreamsUtil() {}
+
+    public static int readFully(InputStream reader, byte[] dest, int offset, int len) throws IOException {
```

Comment:
I was thinking of adding this in a follow up since this PR was intended to be as mechanical of a refactor as possible but I like that it eliminates the `StreamsUtil` class completely so I'll make the change here.